### ISSUE TITLE
Strange new Cantera errors when running distributed

### ISSUE THREAD

**Issue by MTCam** [illinois-ceesd/mirgecom#635]
I and @anderson2981 have both seen this with separate drivers on separate machines.

```
***********************************************************************              |
CanteraError thrown by call_ctml_writer:                                             |
Error executing python while converting input file:                                  |
Python command was: '/p/gpfs1/mtcampbe/CEESD/AutomatedTesting/MIRGE-Timing/timing/em\|
irge.fusion/miniforge3/envs/timing.fusion/bin/python'                                |
exec_stream_t::start: timeout while waiting for child to report via status_pipe      |
Success                                                                              |
[code 0x0000 ()]                                                                     |
***********************************************************************              |
```

One - Four GPUs on one Lassen node worked fine, but I got this error when trying to run on 2 Lassen nodes 4GPUs/node with the weak scaling test described below.

The 4-rank problem size worked ok on 2 nodes - so I suspect the issue is just problem-size-related.

The mesh that worked has (196608) p=3 elements, the one that failed: (393216) elements.

To reproduce (on Lassen):
```
git clone git@github.com:/illinois-ceesd/emirge
cd emirge && ./install.sh --branch=fusion_actx
git clone git@github.com:/illinois-ceesd/drivers_bozzle
cd drivers_bozzle/dist_scale/3d
# get 2 nodes
# lalloc 2 -q pdebug
source ../../../config/activate_env.sh
source ./mirge_batch_env.sh
jsrun -g 1 -a 1 -n 8 bash -c 'POCL_CACHE_DIR=$POCL_CACHE_DIR_ROOT/$$ python -O -m mpi4py ./combozzle.py -i run_config.yaml'
```

Here's what happens when I do it:
https://github.com/illinois-ceesd/drivers_bozzle/blob/main/dist_scale/3d/error_report.txt

**Comment by inducer** [comment]
A few observations:
- There's a segfault in the log prior to the Cantera error.
- The error message speaks of "timeout while waiting for child to report" and says "Python command was"

Both of those suggest that Cantera is trying to fork a subprocess... for reading a CTI file...? Forking in an MPI process is undefined behavior AFAIK, so I'm not surprised there's a crash.

Looking through the Cantera source based on `call_ctml_writer` shows this file:

https://github.com/Cantera/cantera/blob/7879addfa239c617c00763e32d85e58e4bf3b228/src/base/ct2ctml.cpp

which self-describes as a "Driver for the system call to the python executable that converts cti files to ctml files", which corroborates the above theory.

Shot in the dark: Feed it CTML files directly, maybe it won't need to fork?

**Comment by MTCam** [comment]
> A few observations:
> 
> * There's a segfault in the log prior to the Cantera error.
> * The error message speaks of "timeout while waiting for child to report" and says "Python command was"
> 
> Both of those suggest that Cantera is trying to fork a subprocess... for reading a CTI file...? Forking in an MPI process is undefined behavior AFAIK, so I'm not surprised there's a crash.

But this is odd, it works fine on 8 ranks with a slightly smaller mesh.  It works every time, and the slightly larger mesh fails every time with this same error.

> Shot in the dark: Feed it CTML files directly, maybe it won't need to fork?

Thanks, I'm going to try converting now.

**Comment by inducer** [comment]
> But this is odd, it works fine on 8 ranks with a slightly smaller mesh. It works every time, and the slightly larger mesh fails every time with this same error.

Such is undefined behavior. :smiling_imp:

**Comment by MTCam** [comment]
This issue was worked-around by #677 , but may also be resolved by ecisneros8/pyrometheus#51.
